archetypes/embedded/src/main/resources/archetype-resources/src/main/java/Application.java

```diff
@@ -0,0 +1,122 @@
+      Cache<String, Float> stocksCache = SampleCacheContainer.getCache("stockTickers");
+      System.out.println("  Storing key 'RHT' for 10 seconds.");
+      stocksCache.put("RHT", 45.0f, 10, TimeUnit.SECONDS);
+      System.out.printf("  Checking for existence of key.  Is it there? %s\n", stocksCache.containsKey("RHT"));
+      System.out.println("  Sleeping for 10 seconds...");
+      Thread.sleep(10000);
+      System.out.printf("  Checking for existence of key.  Is it there? %s\n", stocksCache.containsKey("RHT"));
+      assert stocksCache.get("RHT") == null;
+   }
+
+   public void asyncOperations() {
+      System.out.println("\n\n3.  Demonstrating asynchronous operations - where writes can be done in a non-blocking fashion.");
+      Cache<String, Integer> wineCache = SampleCacheContainer.getCache("wineCache");
+
+      System.out.println("  Put #1");
+      CompletableFuture<Integer> f1 = wineCache.putAsync("Pinot Noir", 300);
+      System.out.println("  Put #1");
+      CompletableFuture<Integer> f2 = wineCache.putAsync("Merlot", 120);
+      System.out.println("  Put #1");
+      CompletableFuture<Integer> f3 = wineCache.putAsync("Chardonnay", 180);
+
+      // now poll the futures to make sure any remote calls have completed!
+      for (CompletableFuture<Integer> f : Arrays.asList(f1, f2, f3)) {
+         try {
+            System.out.println("  Checking future... ");
+            f.get();
+         } catch (Exception e) {
+            throw new RuntimeException("Operation failed!", e);
+         }
+      }
+      System.out.println("  Everything stored!");
+
+      // TIP: For more examples on using the asynchronous API, visit https://infinispan.org/docs/stable/titles/developing/developing.html#cache_asynchronous_api
+   }
+
+   public void registeringListeners() {
+      System.out.println("\n\n4.  Demonstrating use of listeners.");
+      Cache<Integer, String> anotherCache = SampleCacheContainer.getCache("another");
+      System.out.println("  Attaching listener");
+      MyListener l = new MyListener();
+      anotherCache.addListener(l);
+
+      System.out.println("  Put #1");
+      anotherCache.put(1, "One");
+      System.out.println("  Put #2");
+      anotherCache.put(2, "Two");
+      System.out.println("  Put #3");
+      anotherCache.put(3, "Three");
+
+      // TIP: For more examples on using listeners visit https://infinispan.org/docs/stable/titles/developing/developing.html#listeners_and_notifications
+   }
+
+   public static void main(String[] args) throws Exception {
+      System.out.println("\n\n\n   ********************************  \n\n\n");
+      System.out.println("Hello.  This is a sample application making use of Infinispan.");
+      Application a = new Application();
+      a.basicUse();
+      a.lifespans();
+      a.asyncOperations();
+      a.registeringListeners();
+      System.out.println("Sample complete.");
```

Comment:
It is missing a `CacheManager.stop()` invocation. 
If you use a clustered config and run `mvn exec:java`, the JVM never terminates.

Reply:
Fixed by utilising the `EmbeddedCacheManager` inside a try statement and passing it to `new Application(EmbeddedCacheManager)`.